### PR TITLE
feat:add request check callback

### PR DESCRIPTION
**Pull request by helintongh**
as fowllowing code, can limit client request behavior.

set_body_check 1st parameter is callback, second is response string.
```cpp
  auto f = [](request &req, response &res)->bool {
    int max_body_size = 1024; // 1k
    if (max_body_size < req.body_len()) {
      return false;
    }
    return true;
  };

  server.set_body_check(f, "Request Entity Is Too Large");
```

### CODE REVIEW

**Comment by qicosmos** [comment]
加一下测试。

**Comment by helintongh** [comment]
> 加一下测试。

Add test done. At the same time find a error in coro client, when request body is bigger than limitation server will return 413 code， but client get code is 404，client don't get correct http status code.

File: include/cinatra/response_cv.hpp [hunk 1]
```diff
@@ -403,6 +416,8 @@ inline std::string_view to_string(status_type status) {
       return not_found;
     case status_type::conflict:
       return conflict;
+    case status_type::entity_too_large:
```
[the remark this answers is not in the document]
Comment (helintongh) [reply include/cinatra/response_cv.hpp:419]: > no ut for 413？

coro client get code is 404,don't know why?postman test it that return http code is 413

[the remark this answers is not in the document]
Comment (helintongh) [reply include/cinatra/response_cv.hpp:419]: 413 test 
image

[the remark this answers is not in the document]
Comment (helintongh) [reply include/cinatra/response_cv.hpp:419]: It's can be merged. drogon framework has function setClientMaxBodySize to limit client request max body.We can use this.
image

**Pull request closed without merging**